android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -452,9 +362,154 @@ public void onSignalingChange(PeerConnection.SignalingState signalingState) {
         webRTCModule.sendEvent("peerConnectionSignalingStateChanged", params);
     }
 
+    /**
+     * Triggered when a new track is signaled by the remote peer, as a result of
+     * setRemoteDescription.
+     */
     @Override
     public void onAddTrack(final RtpReceiver receiver, final MediaStream[] mediaStreams) {
         Log.d(TAG, "onAddTrack");
+
+        // Get The timestamp so that we can reorder the transceivers on the JS layer
+        Long tsLong = System.currentTimeMillis()/1000;
+        String ts = tsLong.toString();
+
+        MediaStreamTrack track = receiver.track();
+
+        if (track == null) {
+            return;
+        }
+        if(track.kind().equals(MediaStreamTrack.VIDEO_TRACK_KIND)){
+            videoTrackAdapters.addAdapter((VideoTrack) track);
+        }
+        remoteTracks.put(track.id(), track);
+
+
+        WritableMap params = Arguments.createMap();
+        WritableMap eventInfo = Arguments.createMap();
+        WritableArray streams = Arguments.createArray();
+
+        for (MediaStream stream : mediaStreams) {
+            // Getting the streamReactTag 
+            String streamReactTag = null;
+            for (Map.Entry<String, MediaStream> e : remoteStreams.entrySet()) {
+                if (e.getValue().equals(stream)) {
+                    streamReactTag = e.getKey();
+                    break;
+                }
+            }
+            if (streamReactTag == null) {
+                streamReactTag = UUID.randomUUID().toString();
+                remoteStreams.put(streamReactTag, stream);
+            }
+            streams.pushMap(serializeStream(streamReactTag, stream));
+        }
+
+        eventInfo.putArray("streams", streams);
+        eventInfo.putMap("receiver", serializeReceiver(receiver));
+        eventInfo.putString("timestamp", ts);
+        // Getting the transceiver object associated with the receiver for the event
+        List<RtpTransceiver> transceivers = peerConnection.getTransceivers();
+        for( RtpTransceiver transceiver : transceivers ) {
+            if(transceiver.getReceiver() != null && receiver.id().equals(transceiver.getReceiver().id())) {
+                eventInfo.putMap("transceiver", serializeTransceiver(transceiver));
+                break;
+            }
+        }
+        params.putInt("id", this.id);
+        params.putMap("info", eventInfo);
```

Comment:
Does this add anything? Maybe we could flatten the structure and send all the fields in `params` ?